### PR TITLE
[log4cxx] Add new port

### LINKED ISSUE #6125: Add log4cxx to vcpkg
Yes, I know vcpkg has log4cplus in ports, but if log4cxx was used in the existing code no one will switch to log4cplus.
log4cxx is well documented;
APR dependency is already in ports.

### PR DESCRIPTION
**Pull request by seanyen**
**Describe the pull request**

- What does your PR fix? #6125

- Which triplets are supported/not supported? Have you updated the CI baseline?
  Only enabled `x86-windows` and `x64-windows`, and added the `fail` baselines for other platforms.

The patch is largely borrowed from the [recipe](https://github.com/conda-forge/log4cxx-feedstock/pull/1) I added for `conda-forge`.

### CODE REVIEW

**Comment by seanyen** [comment]
This is ready for review and merge. Thanks!

File: ports/log4cxx/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH})
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+)
+
+vcpkg_install_cmake()
+
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/log4cxx RENAME copyright)
```
Comment (LilyWangL) [inline ports/log4cxx/portfile.cmake:29]: Can you please use command `file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

File: ports/log4cxx/CONTROL [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Source: log4cxx
+Version: 0.10.0-2
+Homepage: https://logging.apache.org/log4cxx/
+Description: Apache log4cxx is a logging framework for C++ patterned after Apache log4j, which uses Apache Portable Runtime for most platform-specific code and should be usable on any platform supported by APR. Apache log4cxx is licensed under the Apache License, an open source license certified by the Open Source Initiative.
+Build-Depends: apr, apr-util
```
Comment (LilyWangL) [inline ports/log4cxx/CONTROL:5]: Can you please add [Supports](https://github.com/microsoft/vcpkg/blob/e62d1361288e83eba786395b60361ab35ba83800/docs/maintainers/control-files.md#supports) for this port?

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by seanyen** [comment]
@LilyWangL Thanks for the feedback. I pushed another iteration for review and merge.

**Comment by seanyen** [comment]
@dan-shaw @LilyWangL Friendly ping. Any actions required from me? I saw the MacOS build is hanging for this pull request. The rest looks green.

**Comment by LilyWangL** [comment]
> @dan-shaw @LilyWangL Friendly ping. Any actions required from me? I saw the MacOS build is hanging for this pull request. The rest looks green.

This is the CI error, it's not related with this PR, please waiting for us fix this.

**Review by LilyWangL — APPROVED** [review]

**Comment by dan-shaw** [comment]
Thanks for the PR!

**Pull request merged**